src/rest-api.js

```diff
@@ -47,8 +52,29 @@ function perform(config = {}, method = "get", path, params = {}) {
 
     query
     .on("success", actions.resolve)
-    .on("error", actions.reject)
-    .on("fail", actions.reject);
+    .on("error", actions.reject);
+
+    if (method === "get") {
```

Comment:
what about firehose batcher calls here ? They won't be retried if we get a 503 error from the API, right ?